test/xpu_api/numerics/complex.number/cases.h

```diff
@@ -218,48 +215,36 @@ CLANG_DIAGNOSTIC_IGNORED_AUTOLOGICAL_CONSTANT_COMPARE
         return non_zero_nan;
     }
     return non_zero;
-
-CLANG_DIAGNOSTIC_POP
 }
 
 inline
 int
 classify(double x)
 {
-// Suppress clang warning: comparison with infinity always evaluates to false in fast floating point modes [-Wtautological-constant-compare]
-CLANG_DIAGNOSTIC_PUSH
-CLANG_DIAGNOSTIC_IGNORED_AUTOLOGICAL_CONSTANT_COMPARE
-
     if (x == 0)
         return zero;
     if (std::isinf(x))
         return inf;
     if (std::isnan(x))
         return NaN;
     return non_zero;
-
-CLANG_DIAGNOSTIC_POP
 }
 
-void is_about(float x, float y)
-{
-// Suppress clang warning: floating-point comparison is always true; constant cannot be represented exactly in type 'float' [-Wliteral-range]
-CLANG_DIAGNOSTIC_PUSH
-CLANG_DIAGNOSTIC_IGNORED_LITERAL_RANGE
-
-    assert(std::abs((x-y)/(x+y)) < 1.e-6);
-
-CLANG_DIAGNOSTIC_POP
-}
+template <typename T>
+constexpr auto __tol = std::numeric_limits<T>::epsilon() * 1e5;
 
-void is_about(double x, double y)
+template <typename T>
+typename std::enable_if<!std::numeric_limits<T>::is_integer, void>::type
+is_about(T x, T y, const T eps = __tol<T>)
 {
-    assert(std::abs((x-y)/(x+y)) < 1.e-14);
+    assert(std::fabs(x - y) <= eps);
 }
 
-void is_about(long double x, long double y)
+template <typename T>
+typename std::enable_if<!std::numeric_limits<T>::is_integer, void>::type
+is_about(const dpl::complex<T>& x, const dpl::complex<T>& y, const T eps = __tol<T>)
 {
-    assert(std::abs((x-y)/(x+y)) < 1.e-14);
+    return is_about(::std::abs(y - x), T(0.), eps);
 }
```

Comment:
The name `is_about` suggests it's a predicate, not an assertion. Also the name sounds somewhat ambiguous, at least to me, I would prefer a more descriptive name, e.g. "is_nearly_equal"

So, please make the name and the semantics match: it should either be `assert_something` for assertion semantics, or `is_something` for a predicate. The predicate variant looks potentially more reusable.

Reply:
It's old function name from original tests.

Reply:
Fine, let's keep the name, but then let's fix the semantics.

By the way, in a different set of libc++ tests in LLVM there is another definition of `is_about`: https://github.com/llvm/llvm-project/blob/main/libcxx/test/std/numerics/numarray/valarray.nonmembers/valarray.transcend/valarray_helper.h. To me, it seems better than the one for complex; maybe we should use something similar to that. And then we could contribute the changes back to LLVM to make the tests better there as well.

Reply:
The current implementation of ```is_about``` function were made by me in https://github.com/oneapi-src/oneDPL/pull/729 after discussion with @MikeDvorskiy about this function. The main goal of changes in this function - to make algorithm of compare two values more clear.